[Backport release-3_10] Fix issues when content cache is used in blocking mode

rldhont opened the pull request:
## Description

Manually backport #35158

Fixes numerous issues when the content cache is used in blocking mode, ultimately resulting in null entries being stored in the cache for remote requests

nyalldawson committed:
Don't use an event loop to wait for task starting

This is already handled in a MUCH nicer way by QgsTask::waitForFinished
(mutex based, not eventloop based)

nyalldawson committed:
Timeout for blocking network requests in content cache should match
the user's preset network timeout value, not a hardcoded value

nyalldawson committed:
Workaround double slot calling causing empty replies to overwrite
valid reply content

rldhont commented:
@nyalldawson, did some changes in QgsTask needs to be backported to fix QgsSvgCache test?

nyalldawson commented:
@rldhont I can't recall - there have been quite a few fixes to QgsTask in the last couple of months which weren't backported (too risky)

rldhont commented:
@nyalldawson I have manually backported  #32838 to get the `QgsTask::waitForFinished` bugfix not automatically backported.

rldhont committed:
Reintroducing and fixing an event loop to wait for task starting

The method QgsTask::waitForFinished doesn't wait the task to be finished if the task has not yet being started (because the task wait to start if there is no thread available for instance).

This bug has been fixed in a much nicer way in version 3.12 and upper, but it has not been backported to version 3.10. If #32838 is backported, for example with #36651, this commit has to be revert.

rldhont commented:
@elpaso I have reintroduced the loop.

nyalldawson commented:
I'm still unhappy about the risk of this change in LTR. Is it really required? Is there any chance you could just defer these changes and not backport?

rldhont commented:
Hi @nyall, the event loop is still in LTR and I have indentify an issue in the code I have wrote when I introduced the blocking mode.

I can't defer. In LTR, QGIS Server in some cases timed out. The loop to wait for task begun can be started even if the task has already begun.

You have removed the event loop in master and 3.12 because @troopa81 has fixed the QgsTask::waitForFinished but his fixed has not been backported in LTR.

So I just revert your event loop removal and fix the way it is started.

nyalldawson commented:
Ok - I'm still not **personally** confident approving this for LTR. I won't stand in the way if someone else wants to take that responsibility themselves and approve this, but I'm not going to be the one who approves :laughing:

rldhont commented:
If no reject this PR, I will merge it and close #36651
